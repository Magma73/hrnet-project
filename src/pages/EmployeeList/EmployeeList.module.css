@import url('../../utils/colors.css');
.main{
    display: flex;
    flex-direction:column;
    align-items: center;
    width: 100%;
    margin-top:20px;
}
.container{
    display: flex;
    flex-direction:column;
    align-items: center;
}
.h1{
    max-width: fit-content;
    text-align: center;
    font-size: 2em;
    color: var(--color-h1);
}
.homeLink{
    margin-top:20px;
}
