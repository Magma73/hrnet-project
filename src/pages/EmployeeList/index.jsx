import styles from "./EmployeeList.module.css";
import { Link } from "react-router-dom"
import TableComponent from "../../components/Table";
import React from "react";


/**
 * Function component EmployeeList - Represent the EmployeeList Page
 * @returns {JSX.Element} The rendered EmployeeList component.
 */
export default function EmployeeList() {
    // console.log(storage);
    return (
        <div>
            <h1 className={styles.h1}>Current Employees</h1>
            <main className={styles.main}>
                <Link to="/">Home</Link>
                <TableComponent />
            </main>
        </div>
    );
};