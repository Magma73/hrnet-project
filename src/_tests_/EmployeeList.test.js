
import * as React from 'react';
import { MemoryRouter, Routes, Route } from 'react-router-dom';
import { create } from 'react-test-renderer';
import { cleanup } from '@testing-library/react';
import EmployeeList from '../pages/EmployeeList';

afterEach(() => {
  cleanup();
});

describe('Given a MemoryRouter with initial entry pointing to the Employee List page', () => {
  test('Then it should render the Employee List page correctly', () => {
    let renderer = create(
      <MemoryRouter initialEntries={['/employeelist']}>
        <Routes>
          <Route path="/employeelist" element={<EmployeeList />} />
        </Routes>
      </MemoryRouter>
    );
    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
