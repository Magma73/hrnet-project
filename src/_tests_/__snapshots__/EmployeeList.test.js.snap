// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a MemoryRouter with initial entry pointing to the Employee List page Then it should render the Employee List page correctly 1`] = `
<div
  className="container"
>
  <h1
    className="h1"
  >
    Current Employees
  </h1>
  <main
    className="main"
  >
    <div
      className="containerTableComponent"
    >
      <div
        className="containerInputs"
      >
        <label>
          Show 
          <select
            id="paginationSelect"
            onChange={[Function]}
            value={10}
          >
            <option
              value={10}
            >
              10
            </option>
            <option
              value={20}
            >
              20
            </option>
            <option
              value={30}
            >
              30
            </option>
            <option
              value={40}
            >
              40
            </option>
            <option
              value={50}
            >
              50
            </option>
          </select>
           entries
        </label>
        <div>
          <label
            htmlFor="globalFilter"
          >
            Search : 
          </label>
          <input
            id="globalFilter"
            onChange={[Function]}
            value=""
          />
        </div>
      </div>
      <table
        className="table"
      >
        <thead>
          <tr
            className="tr"
          />
        </thead>
        <tbody />
      </table>
      <div
        className="containerFooter"
      >
        <div>
          <span>
            Showing 
            1
             to 
            0
             of
             
            0
          </span>
           entries
        </div>
        <div
          className="containerButton"
        >
          <div>
            <button
              disabled={true}
              onClick={[Function]}
            >
              Previous
            </button>
          </div>
          <div>
            <button
              disabled={true}
              onClick={[Function]}
            >
              Next
            </button>
          </div>
        </div>
      </div>
    </div>
    <a
      className="homeLink"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
  </main>
</div>
`;
